modules/session/redis.go

```diff
@@ -127,20 +128,20 @@ func (p *RedisProvider) Init(maxlifetime int64, configs string) (err error) {
 	}
 
 	p.c = nosql.GetManager().GetRedisClient(uri.String())
-	return p.c.Ping().Err()
+	return p.c.Ping(graceful.GetManager().HammerContext()).Err()
```

Comment:
This one should be shutdown

```suggestion
	return p.c.Ping(graceful.GetManager().ShutdownContext()).Err()
```